packages/blocks/src/api/parser.js

```diff
@@ -226,6 +226,10 @@ export function matcherFromSource( sourceConfig ) {
  * @return {*} Attribute value.
  */
 export function parseWithAttributeSchema( innerHTML, attributeSchema ) {
+	if ( attributeSchema.source === 'html' && ! attributeSchema.selector ) {
```

Comment:
Minor: this seems better in the `matcherFromSource` function because there's already a switch case there.

Reply:
`matcherFromSource` always gets fed into `hpq` and will then get modified. The idea with the change is to bypass `hpq` entirely, for which `parseWithAttributeSchema` seems a good place:

```js
export function parseWithAttributeSchema( innerHTML, attributeSchema ) {
	if ( attributeSchema.source === 'html' && ! attributeSchema.selector ) {
		return innerHTML;
	}

	return hpqParse( innerHTML, matcherFromSource( attributeSchema ) );
}
```